torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -997,6 +997,19 @@ def rotate_image(
     center: Optional[List[float]] = None,
     fill: _FillTypeJIT = None,
 ) -> torch.Tensor:
+    angle = angle % 360  # shift angle to [0, 360) range
+
+    # fast path: transpose without affine transform
+    if center is None:
```

Comment:
The tests are failing and I think it's because we should set `expand` to True. When `expand` is False we're not supposed to be changing the shape of the output (which is what `rot90` does!).  

```suggestion
    if expand and center is None:
```

Reply:
Here is what does PIL here:
https://github.com/python-pillow/Pillow/blob/8f63748e50378424628155994efd7e0739a4d1d1/src/PIL/Image.py#L2287-L2296

Yes, I agree that it is incorrect to omit expand for image with h != w, sorry for an incorrect suggestion previously.

Reply:
Should we also check for square input then? So even if expand is false, rot90 is still valid.

Reply:
Following their code they: 1) accept angle 0 and 180 whatever provided expand value and 2) for rotations 90 or 270 they do `if angle in (90, 270) and (expand or self.width == self.height)`.
I think this is reasonable and we can do the same

Reply:
I will update accordingly

Reply:
I have updated the code. Also updated the test cases to cover everything. I just realized the tests didn't fail on my machine because I was running inside an environment with torchvision installed (big mistake!).

Reply:
No worries @gau-nernst thanks for the update. I can confirm the tests are passing now (tried locally). let's just wait for the CI to be green before merging